sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java

```diff
@@ -277,28 +258,26 @@ private void exportBatches(ArrayList<ReadableSpan> spanList) {
     }
 
     // Exports the list of SpanData to the SpanExporter.
+    @SuppressWarnings("BooleanParameter")
     private void onBatchExport(final List<SpanData> spans) {
-      Future<?> submission =
-          executorService.submit(
+      if (exportAvailable.compareAndSet(true, false)) {
```

Comment:
IIUC this changes the behavior to export essentially being "scheduleAtFixedRate" instead of the current "scheduleAtFixedDelay" since this always returns immediately now. I don't know if this is the intention of this class. Also given a significant refactor in #1395 which will make this class much easier to reason about, maybe we can keep the behavior unchanged for now?

Reply:
I think we still need the changes here. Contractually, we should guarantee that an export function is not called if it has a request in-flight. It isn't yet clear to me how the proposed changes of #1395 would help.

Reply:
Unfortunately, this change from "scheduleAtFixedDelay" to "scheduleAtFixedRate" results in dropped spans. `onBatchExport` can be called in a loop from `exportBatches` method. As this method now returns immediately, calling it in a tight loop guarantees data loss.

Reply:
> Unfortunately, this change from "scheduleAtFixedDelay" to "scheduleAtFixedRate" results in dropped spans. `onBatchExport` can be called in a loop from `exportBatches` method. As this method now returns immediately, calling it in a tight loop guarantees data loss.

I think you're right - for situations where the `spansList` cannot be drained fast enough. I shall attend to this. Meanwhile, here's a new ticket: https://github.com/open-telemetry/opentelemetry-java/issues/1566 to capture the problem. It isn't a terrible situation as `BatchSpanProcessor` usage has always dictated that spans can be dropped. However, we can do better.